File: example implementations/java/convertor/src/io/CvoxMultimodel.java
package io;

import utils.EL;
import voxel.VoxModel;
import voxel.XYZ;

public class CvoxMultimodel {
    public EL<CVoxModel> models = new EL<>();
    public EL<XYZ> translations = new EL<>();
    public XYZ size = new XYZ();
    static XYZ modelLimit = new XYZ(256);

    public CvoxMultimodel(){}

    public CvoxMultimodel(VoxModel voxModel) {
        size = voxModel.size;
        VoxModel[][][] voxModels = voxModel.simpleSplit(modelLimit);
        for (int xx = 0; xx < voxModels.length; xx++)
            for (int yy = 0; yy < voxModels[xx].length; yy++)
                for (int zz = 0; zz < voxModels[xx][yy].length; zz++)
                    add(new CVoxModel(voxModels[xx][yy][zz]), new XYZ(xx, yy, zz).mult(modelLimit));
    }

    public void add(CVoxModel model, XYZ translation) {
        models.add(model);
        translations.add(translation);
        size = size.max(model.size.add(translation));
    }
}

File: example implementations/csharp/cvox-convertor/io/CvoxMultimodel.cs
using cvox_convertor.voxel;

namespace cvox_convertor.io
{
    public class CvoxMultimodel
    {
        public List<CvoxModel> Models = new();
        public List<XYZ> Translations = new();
        public XYZ Size = new();
        static readonly XYZ ModelLimit = new(256);

        public CvoxMultimodel() { }

        public CvoxMultimodel(VoxModel voxModel)
        {
            Size = voxModel.Size;
            VoxModel[,,] voxModels = voxModel.SimpleSplit(ModelLimit);
            for (int xx = 0; xx < voxModels.GetLength(0); xx++)
                for (int yy = 0; yy < voxModels.GetLength(1); yy++)
                    for (int zz = 0; zz < voxModels.GetLength(2); zz++)
                        Add(new CvoxModel(voxModels[xx, yy, zz]), new XYZ(xx, yy, zz) * ModelLimit);
        }

        public void Add(CvoxModel model, XYZ translation)
        {
            Models.Add(model);
            Translations.Add(translation);
            Size = Size.Max(model.size + translation);
        }
    }
}
